src/Analyzer.Utilities/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -93,8 +93,18 @@ public static IEnumerable<ITypeSymbol> GetBaseTypesAndThis(this ITypeSymbol type
             }
         }
 
-        public static bool DerivesFrom(this INamedTypeSymbol symbol, INamedTypeSymbol candidateBaseType)
+        public static bool DerivesFrom(this ITypeSymbol symbol, ITypeSymbol candidateBaseType, bool baseTypesOnly = false)
```

Comment:
Having this extension method on a more concrete type while not using any specific info in the subclass is not appropriate. If everything needed at the ITypeSymbol level, then it is possbile to do this on this level, and there will be more extensive client code being able to leverage this. Generally speaking, this makes this extension more applicable.